### PR TITLE
Remove interstitial

### PR DESCRIPTION
**Pull request by smlovin2**
- Removes the pieces show page, helper and corresponding tests
- Adds jquery script that makes the pieces draggable and spaces droppable
- Adds capybara tests to test out drag and drop functionality
- Adds a few new gems and code to help out with capybara tests
- Adds code to pieces controller to check for valid move before updating the pieces attributes
- Does some cleaning up of the css

**Concerns:**
- Mike there is a bit of this that I am not sure about, like setting async to false for the ajax request and the render stuff in the controller so if you if you know of any better ways to accomplish I would be more than happy to hear about it :smile:

### CODE REVIEW

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-    current_piece.update_attributes(pieces_params)
-    redirect_to game_path(current_piece.game)
+    row = pieces_params[:row].to_i
+    column = pieces_params[:column].to_i
+
+    if current_piece.valid_move?(row, column)
+      current_piece.update_attributes(pieces_params)
+      render text: 'updated!'
+    else
+      render text: 'invalid', status: :bad_request
```
Comment (smlovin2) [inline app/controllers/pieces_controller.rb]: @mscoutermarsh Wasn't sure what to do exactly for the render stuff. I have it this way so that the ajax request in the javascript will actually hit the .done() function. Is there another way to only return the http status and still trigger the ajax.done() function?

Also, I know I asked you earlier today but are we going to eventually just be calling move_to! here and let it take care of updating the attributes or returning false for an invalid move?

File: app/views/games/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,29 @@
+<script>
+  $(function() {
+    $('.piece').draggable();
+    $('td').droppable({
+      drop: function(ev, ui) {
+        var valid_move = false;
+
+        $.ajax({
+          type: 'PUT',
+          async: false,
```
Comment (smlovin2) [inline app/views/games/show.html.erb]: @mscoutermarsh I put this here so that the valid_move variable would be updated before doing the if statement later. From the stuff I read though this seems like a bad thing since that kind of defeats the purpose of ajax. I had tried putting the `ui.draggable.draggable('option', 'revert', true);` stuff in a ajax.fail() function but it wasn't working that way. Do you have any suggestions on a better way to do this?

**Comment by smlovin2** [comment]
@mscoutermarsh Is there a way to see what the conflicts here on GitHub or can you only see that when you use the command line? I'm just curious to see what files are causing the problem.

File: app/controllers/pieces_controller.rb (hunk 1 above)
Comment (mscoutermarsh) [inline app/controllers/pieces_controller.rb]: Yes, we'll be calling `move_to!` here.

If you want to add it now. Go for it. Or it could be a separate pull request.

Re: http status.

This should also work

``` Ruby
## For good
render status: :ok

## For bad
render status: :bad_request
```

File: app/views/games/show.html.erb (hunk 2 above)
Comment (mscoutermarsh) [inline app/views/games/show.html.erb]: hmm I would expect `fail()` to work.

If your rails controller returns a `success` status. such as `render status: :ok`. Then it should run the `done` code.

If it fails, have it return `render status: :bad_request`. And it should run the `fail` code.

I haven't used jquery in a while. Here is some example code I stole.

``` JavaScript
var subscribeRequest = $.ajax({
     type: "POST",
     url: "subscribe.php",
     data: { subscriberName: $('#name').val(), emailAddress: $('#email').val() }
});

subscribeRequest.done(function(msg)
{
     alert( "You have successfully subscribed to our mailing list." );
});

subscribeRequest.fail(function(jqXHR, textStatus)
{
     alert( "We could not subscribe you please try again or contact us if the problem persists (" + textStatus + ")." );
});
```

from: http://www.sitepoint.com/5-jquery-ajax-examples/

Comment (mscoutermarsh) [inline app/views/games/show.html.erb]: (sounds like this might be what you already tried though)

File: spec/features/move_pieces_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  given(:game) { FactoryGirl.create(:game) }
+
+  scenario 'Dragging a piece(black_pawn) to a valid location' do
+    game.pieces << FactoryGirl.create(:black_pawn, type: 'Pawn')
+    visit game_path(game)
+    black_pawn = page.find('.ui-draggable', text: 'o')
+    destination = page.find('tr:nth-child(3) td:nth-child(5)')
+    black_pawn.drag_to(destination)
```
Comment (mscoutermarsh) [inline spec/features/move_pieces_spec.rb:15]: wow, cool.

**Comment by mscoutermarsh** [comment]
@smlovin2 you crushed this. it looks great. :raised_hands: 

If you can get the jquery to use the built in `done/fail`. That would be ideal. But this works, so I wouldn't spend a ton of time on it.

For the merge conflicts, only way to see them is to do the merge via command line (or some GUI git tool).
